### PR TITLE
Improve error handling and debugging experience

### LINKED ISSUE #344: Improve the debugging experience of LitServe
Right now, it is challenging to debug when the server is failing.

This arises from 2 main reasons:

1. It is not possible to put a breakpoint within the server
2. The error are quite critic and hard to follow.

Example: I copied a server from the docs and put together a client and got this. Quite unclear where the error is originated from.

```python
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.10/site-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.10/site-packages/starlette/_exception_handler.py", line 62, in wrapped_app
    raise exc
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.10/site-packages/starlette/_exception_handler.py", line 51, in wrapped_app
    await app(scope, receive, sender)
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.10/site-packages/starlette/routing.py", line 73, in app
    response = await f(request)
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.10/site-packages/fastapi/routing.py", line 301, in app
    raw_response = await run_endpoint_function(
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.10/site-packages/fastapi/routing.py", line 212, in run_endpoint_function
    return await dependant.call(**values)
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.10/site-packages/litserve/server.py", line 644, in predict
    load_and_raise(response)
  File "/home/zeus/miniconda3/envs/cloudspace/lib/python3.10/site-packages/litserve/utils.py", line 41, in load_and_raise
    raise exception
RuntimeError: a Tensor with 2 elements cannot be converted to Scalar
```

This was happening within the `encode_response` but directed me to the predict.

Here are 2 potential improvements direction to ease adoption:

1. This breakpoint enables to debug in LitServe

```python
import multiprocessing
import os
import pdb
import sys

_stdin = [None]
_stdin_lock = multiprocessing.Lock()
try:
    _stdin_fd = sys.stdin.fileno()
except Exception:
    _stdin_fd = None


# Taken from https://github.com/facebookresearch/metaseq/blob/main/metaseq/pdb.py
class MPPdb(pdb.Pdb):
    """A Pdb wrapper that works in a multiprocessing environment."""

    def __init__(self) -> None:
        pdb.Pdb.__init__(self, nosigint=True)

    def _cmdloop(self) -> None:
        stdin_back = sys.stdin
        with _stdin_lock:
            try:
                if _stdin_fd is not None:
                    if not _stdin[0]:
                        _stdin[0] = os.fdopen(_stdin_fd)
                    sys.stdin = _stdin[0]
                self.cmdloop()
            finally:
                sys.stdin = stdin_back


def set_trace() -> None:
    pdb = MPPdb()
    pdb.set_trace(sys._getframe().f_back)
```

2. Capturing the cleaned error trace would ease debugging  

----

### PR DESCRIPTION
**Pull request by aniketmaurya**
## What does this PR do?

User's terminal will have clear error logs for debugging 

fixes #344

## Example:

```python
import litserve as ls

class EmbeddingsAPI(ls.LitAPI):
    def setup(self, device):
        self.model = SentenceTransformer('all-MiniLM-L6-v2', device=device)

    def predict(self, inputs):
        raise Exception("internal server error")

    def encode_response(self, output, **kwargs):
        return {"embeddings": output}

if __name__ == "__main__":
    api = EmbeddingsAPI()
    server = ls.LitServer(api, spec=ls.OpenAIEmbeddingSpec(), max_batch_size=4)
    server.run(port=8000)
```

### Before (cryptic ASGI logs)
image

### After (clear errors)
image

<details>
  <summary><b>Before submitting</b></summary>

- [ ] Was this discussed/agreed via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/pytorch-lightning/blob/main/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

</details>

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in GitHub issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by tchaton** [comment]
Nit: For readability, there is usually a space after the `: `

Screenshot 2024-12-09 at 16 29 49

**Review by tchaton — APPROVED** [review]

**Review by robTheBuildr — APPROVED** [review]

**Review by Andrei-Aksionov — APPROVED** [review]
